src/test/java/com/limechain/sync/warpsync/state/RuntimeDownloadTest.java

```diff
@@ -0,0 +1,106 @@
+package com.limechain.sync.warpsync.state;
+
+import com.limechain.network.kad.KademliaService;
+import com.limechain.network.protocol.lightclient.LightMessages;
+import com.limechain.network.protocol.lightclient.LightMessagesProtocol;
+import com.limechain.network.protocol.lightclient.pb.LightClientMessage;
+import com.limechain.sync.warpsync.runtime.RuntimeBuilder;
+import com.limechain.trie.Trie;
+import com.limechain.trie.TrieVerifier;
+import com.limechain.trie.decoder.TrieDecoderException;
+import com.limechain.utils.LittleEndianUtils;
+import com.limechain.utils.StringUtils;
+import io.emeraldpay.polkaj.scale.ScaleCodecReader;
+import io.emeraldpay.polkaj.types.Hash256;
+import io.ipfs.multihash.Multihash;
+import io.libp2p.core.Host;
+import io.libp2p.core.PeerId;
+import io.libp2p.protocol.Ping;
+import lombok.extern.java.Log;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.peergos.HostBuilder;
+
+import java.util.List;
+import java.util.Random;
+import java.util.logging.Level;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+@Log
+public class RuntimeDownloadTest {
+    @Disabled("This is an integration test")
+    @Test
+    public void runtimeDownloadAndBuildTest() {
+        Host senderNode = null;
+        try {
+            HostBuilder hostBuilder1 =
+                    (new HostBuilder()).generateIdentity().listenLocalhost(10000
+                            + new Random().nextInt(50000));
+
+            var lightMessages = new LightMessages("/dot/light/2", new LightMessagesProtocol());
+            var kademliaService = new KademliaService("/dot/kad",
+                    Multihash.deserialize(hostBuilder1.getPeerId().getBytes()), false, true);
+
+            hostBuilder1.addProtocols(List.of(new Ping(), lightMessages, kademliaService.getProtocol()));
+            senderNode = hostBuilder1.build();
+
+            senderNode.start().join();
+
+            kademliaService.setHost(senderNode);
+            var peerId = PeerId.fromBase58("12D3KooWFFqjBKoSdQniRpw1Y8W6kkV7takWv1DU2ZMkaA81PYVq");
+            var receivers = new String[]{
+//                    "/ip4/127.0.0.1/tcp/30333/p2p/12D3KooWPyKTVdykB9iEXEggRkbMTV4qsWywhtT42qz16eBTvReA"
```

Comment:
This can be deleted if it's not in use. Also some comments in this test would be nice as the code is not very easy to read.